lib/screens/profile/contact_support_public.dart

```diff
@@ -0,0 +1,292 @@
+      final rows = deviceInfo?.entries.map((entry) => '<th>${entry.key}</th><td>${entry.value}</td>').join('\n');
+      return '<table><tbody>$rows</tbody></table>';
+    }
+
+    Future<QueryResult> getSendSupportEmailFuture() {
+      final appVer = deviceInfo?.appVer;
+      final os = deviceInfo?.os;
+      return ref.read(gqlClientProvider).mutate$sendSupportEmail(
+            Options$Mutation$sendSupportEmail(
+              variables: Variables$Mutation$sendSupportEmail(
+                title: 'BTV $appVer $os',
+                content: '',
+                html: '<p>${messageController.text}</p>${includeDeviceInfo.value ? getDeviceInfoHtml() : ''}',
+                options: Input$EmailOptions(name: nameController.text, email: emailController.text),
+              ),
+            ),
+          );
+    }
+
+    void onSend() {
+      FocusManager.instance.primaryFocus?.unfocus();
+
+      if (formKey.currentState?.validate() != true) {
+        return;
+      }
+
+      sendSupportEmailFuture.value = getSendSupportEmailFuture();
+      isOnInputPage.value = false;
+    }
+
+    final Widget body;
+    if (isOnInputPage.value) {
+      body = Form(
+        key: formKey,
+        child: _InputPage(
+          nameController: nameController,
+          emailController: emailController,
+          messageController: messageController,
+          onIncludeDeviceInfoChange: onIncludeDeviceInfoChange,
+        ),
+      );
+    } else if (sendSupportEmailSnapshot.hasError || sendSupportEmailSnapshot.data?.hasException == true) {
+      body = ContactSupportFailure(onTryAgain: onTryAgain);
+    } else if (sendSupportEmailSnapshot.hasData) {
+      body = const ContactSupportSuccess();
+    } else {
+      body = const ContactSupportSendingIndicator();
+    }
+
+    return DialogOnWeb(
+      child: GestureDetector(
+        onTap: () => FocusManager.instance.primaryFocus?.unfocus(),
+        child: Scaffold(
+          appBar: GeneralAppBar(
+            leftActions: [
+              BtvButton(
+                labelText: S.of(context).cancel,
+                onPressed: context.router.pop,
+              )
+            ],
+            rightActions: [
+              AnimatedSwitcher(
+                duration: const Duration(milliseconds: 150),
+                child: isOnInputPage.value && messageController.text.isNotEmpty && nameController.text.isNotEmpty && emailController.text.isNotEmpty
+                    ? DesignSystem.of(context).buttons.small(
+                          labelText: S.of(context).send,
+                          onPressed: onSend,
+                        )
+                    : null,
+              )
+            ],
+          ),
+          body: SafeArea(
+            child: AnimatedSwitcher(
+              duration: const Duration(milliseconds: 200),
+              child: body,
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class _InputPage extends HookWidget {
```

Comment:
_InputPage is the only "body" which didn't get its own file. It would make more sense the other way around, since inputpage is the primary widget and the other ones are very small and.
Consider moving the loading indicator as a private component in here and using errorgeneric.

Reply:
The reason I moved those components to separate files was more for reuse because they were used in both the public screen and the normal screen.